Doctrine/Orm/Filter/SearchFilter.php

```diff
@@ -59,10 +55,48 @@ public function __construct(
         PropertyAccessorInterface $propertyAccessor,
         array $properties = null
     ) {
-        $this->managerRegistry = $managerRegistry;
+        parent::__construct($managerRegistry, $properties);
+
         $this->iriConverter = $iriConverter;
         $this->propertyAccessor = $propertyAccessor;
-        $this->properties = $properties;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function apply(ResourceInterface $resource, QueryBuilder $queryBuilder, Request $request)
+    {
+        $metadata = $this->getClassMetadata($resource);
+        $fieldNames = array_flip($metadata->getFieldNames());
+
+        foreach ($this->extractProperties($request) as $property => $value) {
+            if (!is_string($value) || !$this->isPropertyEnabled($property)) {
+                continue;
+            }
+
+            $partial = null !== $this->properties && self::STRATEGY_PARTIAL === $this->properties[$property];
+
+            if (isset($fieldNames[$property])) {
+                if ('id' === $property) {
+                    $value = $this->getFilterValueFromUrl($value);
+                }
+
+                $queryBuilder
+                    ->andWhere(sprintf('o.%1$s LIKE :%1$s', $property))
```

Comment:
I'm still a little concerned about the parameter name too. The filters are called without much control and heavily modify the queryBuilder, hence should be isolated as much as possible. Prefixing the parameter to avoid any conflict won't hurt IMO.